File: backend/F1StatsServer/Model/Track.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using F1StatsServer.Util;
using Microsoft.EntityFrameworkCore;

namespace F1StatsServer.Model;

[Table("Track")]
public partial class Track : EntityBase
{
    [StringLength(255)]
    [Unicode(false)]
    public string Name { get; set; } = null!;

    public byte? CornersTotal { get; set; }

    public byte? CornersLeft { get; set; }

    [Column(TypeName = "decimal(5, 1)")]
    public decimal? Elevation { get; set; }

    public short? Length { get; set; }

    public byte? PitStop { get; set; }

    [StringLength(255)]
    [Unicode(false)]
    public string? ImagePath { get; set; }

    public byte Laps { get; set; }

    public virtual ICollection<GrandPrix> GrandPrixes { get; set; } = new List<GrandPrix>();

    [ForeignKey("TrackId")]
    [InverseProperty("Tracks")]
    public virtual ICollection<Country> Countries { get; set; } = new List<Country>();

}

File: frontend/src/app/shared/models/track/Track.ts
export class Track {
    id: number = 0;
    name: string = "Albert Park Circuit";
    cornersTotal: number = 14;
    cornersLeft: number = 6;
    elevation: number = 2.4;
    length: number = 5278;
    pitStop?: number | null = 25;
    imagePath?: string | null;
    countryId: number = 10;
    location: string = "Melbourne";
}